dht_net.go

```diff
@@ -300,12 +310,17 @@ func (dht *IpfsDHT) messageSenderForPeer(ctx context.Context, p peer.ID) (*messa
 	return ms, nil
 }
 
+type protocolSender interface {
```

Comment:
is this only used for the `NewStream` construction below? if so, might be easier to set the interface to a function returning a stream given a peerID?